AOC3/src/main/kotlin/Day16.kt

```diff
@@ -43,9 +40,8 @@ fun main() {
 
 fun spinDanceMove(positionNumber: Int) {
 
-    val indexSpinDanceMove = programsChar.size - positionNumber
-    val finalSlicedPart = programsChar.slice(indexSpinDanceMove until programsChar.size)
-    val startingSlicedPart = programsChar.slice(0 until indexSpinDanceMove)
+    val finalSlicedPart = programsChar.takeLast(positionNumber)
+    val startingSlicedPart = programsChar.take(programsChar.size - positionNumber)

```

Comment:
Again, these `String` methods make this much easier. There are even better ways, but I stayed with the relatively easy to implement given your solution.

You can take the same approach here in the other functions below.